clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java

```diff
@@ -0,0 +1,171 @@
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.EcsCloudProvider;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsInstance;
+import com.netflix.spinnaker.clouddriver.model.InstanceProvider;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Component
+public class EcsInstanceProvider implements InstanceProvider<EcsInstance> {
+
+  private final String cloudProvider = EcsCloudProvider.ID;
+
+  @Autowired
+  private AccountCredentialsProvider accountCredentialsProvider;
+
+  @Autowired
+  private AmazonClientProvider amazonClientProvider;
+
+
+  @Override
+  public String getCloudProvider() {
+    return cloudProvider;
+  }
+
+  @Override
+  public EcsInstance getInstance(String account, String region, String id) {
+    if (!isValidId(id, region))
+      return null;
+
+    EcsInstance ecsInstance = null;
+
+    AWSCredentialsProvider awsCredentialsProvider = getCredentials(account).getCredentialsProvider();
+    AmazonECS amazonECS = amazonClientProvider.getAmazonECS(account, awsCredentialsProvider, region);
+    AmazonEC2 amazonEC2 = amazonClientProvider.getAmazonEC2(account, awsCredentialsProvider, region);
+
+    Task ecsTask = getTask(amazonECS, id);
+    InstanceStatus instanceStatus = getEC2InstanceStatus(amazonEC2, getContainerInstance(amazonECS, ecsTask));
+
+    if (ecsTask != null && instanceStatus != null) {
+      ecsInstance = new EcsInstance(id, ecsTask, instanceStatus);
+    }
+
+    return ecsInstance;
+  }
+
+  @Override
+  public String getConsoleOutput(String account, String region, String id) {
+    return null;
+  }
+
+  private List<String> getAllClusters(AmazonECS amazonECS) {
+    ListClustersResult listClustersResult = amazonECS.listClusters();
+    List<String> clusterList = listClustersResult.getClusterArns();
+    while (listClustersResult.getNextToken() != null) {
+      listClustersResult = amazonECS.listClusters(
+        new ListClustersRequest().withNextToken(listClustersResult.getNextToken())
+      );
+      clusterList.addAll(listClustersResult.getClusterArns());
+    }
+    return clusterList;
+  }
+
+  private NetflixAmazonCredentials getCredentials(String account) {
+    NetflixAmazonCredentials accountCredentials = null;
+    for (AccountCredentials credentials: accountCredentialsProvider.getAll()) {
+      if (credentials.getName().equals(account)) {
+        if (credentials instanceof NetflixAmazonCredentials) {
+          accountCredentials = (NetflixAmazonCredentials) credentials;
+          break;
```

Comment:
Instead of breaking the loop, you can return accountCredentials.  If it is an instance of NetflixAmazonCredentials, then it is not null, then it can be returned

Reply:
null instanceof NetflixAmazonCredentials will not cause an NPE btw.  As far as I can remember :)